src/device-registry/models/Device.js

```diff
@@ -462,7 +462,14 @@ deviceSchema.statics = {
         .allowDiskUse(true);
 
       if (Object.keys(exclusionProjection).length > 0) {
-        pipeline.project(exclusionProjection);
+        // Construct the $project stage dynamically with the exclusionProjection
```

Comment:
Hi @BenjaminSsempala , please take note of this runtime error that has just been fixed and apply the same fix in `auth-service` in case it is also needed there? Thanks!
Reference can be made to the numerous Slack notifications inside `#notifs-device-registry`